qiskit/compiler/transpile.py

```diff
@@ -201,28 +207,72 @@ def callback_func(**kwargs):
                                   'is greater than maximum ({}) '.format(max_qubits) +
                                   'in the coupling_map')
     # Transpile circuits in parallel
-    circuits = parallel_map(_transpile_circuit, list(zip(circuits, transpile_configs)))
+    circuits = parallel_map(_transpile_circuit, list(zip(circuits, transpile_args)))
 
     if len(circuits) == 1:
         return circuits[0]
     return circuits
 
 
-# FIXME: This is a helper function because of parallel tools.
 def _transpile_circuit(circuit_config_tuple):
     """Select a PassManager and run a single circuit through it.
-
     Args:
         circuit_config_tuple (tuple):
             circuit (QuantumCircuit): circuit to transpile
-            transpile_config (TranspileConfig): configuration dictating how to transpile
-
+            transpile_config (dict): configuration dictating how to transpile. The format of the
+                dictionary has the following format:
+                {'optimization_level': int,
+                 'pass_manager': PassManager,
+                 'output_name': string,
+                 'callback': callable,
+                 'pass_manager_config': PassManagerConfig}
```

Comment:
make sure all docstrings are accurate. this one still has optimization_level and pass_manager_config in it.

Reply:
Because this is `_transpile_circuit`, not `transpile_circuit`. Changing it here implies modifying `_parse_transpile_args`. It is necessary, but this PR is already hard to understand...

Reply:
ok, can you rename `transpile_config` -> `transpile_args` then to avoid confusion with the deprecated class of the same name?